test/runify.js

```diff
@@ -339,6 +339,52 @@ hosts.forEach(function (record) {
       });
     });
 
+    it('tolerates broken execution environments', function () {
+      return agent.evalScript(`
+            Object.defineProperty(Object.prototype, "length", {
+                get: function () {
+                    return 1;
+                },
+                configurable: true
+              });
+
+              print('okay');
+          `).then(function(result) {
+             assert.equal(result.stderr, '');
+             assert(result.stdout.match(/^okay\r?\n/m));
+            });;
+    });
+
+    it('supports realm nesting', function () {
+      return agent.evalScript(`
+           x = 0;
+           $.createRealm().evalScript(\`
+             x = '';
+             $.createRealm().evalScript(\\\`
+               x = {};
+               print(typeof x);
+             \\\`);
+             print(typeof x);
+           \`);
+           print(typeof x);
+         `)
+        .then(function(result) {
+            assert.equal(result.stderr, '');
+            assert(result.stdout.match(/^object\r?\nstring\r?\nnumber\r?\n/m));
+          });
+    });
+
+    it('observes correct cross-script interaction semantics', function () {
+      return agent.evalScript(`
+           print($.evalScript('let eshost;').type);
+           print($.evalScript('let eshost;').type);
+         `)
+        .then(function(result) {
```

Comment:
The file is largely inconsistent in this regard, but I'm happy to use arrows here if you prefer.